build/v1/types.go

```diff
@@ -1358,12 +1362,15 @@ type BuildVolume struct {
 	// Names that collide with those added by the build controller will result in a
 	// failed build with an error message detailing which name caused the error.
 	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
+	// +required
 	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
 
 	// source represents the location and type of the mounted volume.
+	// +required
 	Source BuildVolumeSource `json:"source" protobuf:"bytes,2,opt,name=source"`
 
 	// mounts represents the location of the volume in the image build container
+	// +listType=atomic
```

Comment:
just to be sure: you would expect that this is replaced completely instead of being merged?

Reply:
this is updated now, let me know if that makes sense.